content/en/platform/corda/5.0/developing-applications/basic-cordapp/contract.md

```diff
@@ -0,0 +1,293 @@
+                    "This transaction should only have one AppleStamp state as output"
+                }
+                require(output.stampDesc.isNotBlank()) {
+                    "The output AppleStamp state should have clear description of the type of redeemable goods"
+                }
+            }
+            is AppleCommands.Redeem -> {
+                // Transaction verification will happen in BasketOfApplesContract
+            }
+            else -> {
+                // Unrecognised Command type
+                throw IllegalArgumentException("Incorrect type of AppleStamp commands: ${command::class.java.name}")
+            }
+        }
+    }
+}
+
+```
+
+## Create the `BasketOfApplesContract`
+
+The `BasketOfApplesContract` has two intentions:
+
+* Farmer Bob creates the basket of apples. This intention is expressed by the `PackBasket` command.
+* Dave redeems the `BasketOfApples` state. This intention is expressed by the `Redeem` command, which you defined when creating `AppleStampContract`.
+
+The rules inside the `verify` method in the `requireThat` Corda DSL helper method are:
+
+* For the `PackBasket` command:
+  * This transaction should only output one `BasketOfApples` state.
+  * The output of the `BasketOfApples` state should have a clear description of the apple product.
+  * The output of the `BasketOfApples` state should have a non-zero weight.
+* For the `Redeem` command:
+  * The transaction should consume exactly two states: one `AppleStamp` and one `BasketOfApples`.
+  * The issuer of the `AppleStamp` should be the producing farm of this basket of apples.
+  * The weight of the basket of apples must be greater than zero.
+
+Using what you learned when writing `AppleStampContract`, implement the above in a new class. You will also need to
+update the `AppleCommands` interface defined earlier to account for the new `PackBasket` command.
+
+### Check Your Work
+
+Once you have written the `BasketOfApplesContract`, check your code against the samples below.
+
+Your `AppleCommands` file should look like this:
+
+```kotlin
+package com.r3.developers.apples.contracts
+
+import net.corda.v5.ledger.utxo.Command
+
+// Used to indicate the transaction's intent
+interface AppleCommands : Command {
+    class Issue : AppleCommands
+    class Redeem : AppleCommands
+    class PackBasket : AppleCommands
+}
+```
+
+Your BasketOfApplesContract file should look like this:
+
+```kotlin
+package com.r3.developers.apples.contracts
+
+import com.r3.developers.apples.states.AppleStamp
+import com.r3.developers.apples.states.BasketOfApples
+import net.corda.v5.ledger.utxo.Contract
+import net.corda.v5.ledger.utxo.transaction.UtxoLedgerTransaction
+
+
+class BasketOfApplesContract : Contract{
+
+    override fun verify(transaction: UtxoLedgerTransaction) {
+        // Extract the command from the transaction
+        when (val command = transaction.commands.first()) {
+            is AppleCommands.PackBasket -> {
```

Comment:
no case for `Issue` command

Reply:
@yarivus that's intentional, because that command is only valid in the context of `AppleStamp`, not `BasketOfApples`

Reply:
I hope it won't confuse the CorDapp developers following the documentation ...
Maybe worth adding a comment/note on this ?

Reply:
Added as a suggested change above.

Reply:
Change committed.